### PR TITLE
[REF] mail: move discuss.channel code to its own folder (controllers)

### PR DESCRIPTION
**Pull request by seb-odoo**
This commit focuses on removing all references to discuss.channel from controllers of mail module.

In preparation of splitting discuss and mail modules.

Part of task-3265211

### CODE REVIEW

File: addons/mail/controllers/discuss/binary.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    @http.route(
+        "/discuss/channel/<int:channel_id>/attachment/<int:attachment_id>", methods=["GET"], type="http", auth="public"
+    )
+    def discuss_channel_attachment(self, channel_id, attachment_id, download=None):
+        channel_member_sudo = request.env["discuss.channel.member"]._get_as_sudo_from_request_or_raise(
+            request=request, channel_id=int(channel_id)
+        )
+        domain = [
+            ("id", "=", int(attachment_id)),
+            ("res_id", "=", int(channel_id)),
```
Comment (oomsveta) [inline addons/mail/controllers/discuss/binary.py:70]: the ids are already cast in the route, aren't they?

File: addons/mail/controllers/discuss/binary.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    @http.route(
+        [
+            "/discuss/channel/<int:channel_id>/image/<int:attachment_id>",
+            "/discuss/channel/<int:channel_id>/image/<int:attachment_id>/<int:width>x<int:height>",
+        ],
+        methods=["GET"],
+        type="http",
+        auth="public",
+    )
+    def fetch_image(self, channel_id, attachment_id, width=0, height=0, **kwargs):
```
Comment (oomsveta) [inline addons/mail/controllers/discuss/binary.py:87]: no `discuss` prefix for this one?

File: addons/mail/controllers/discuss/rtc.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+from collections import defaultdict
+
+from odoo import http
+from odoo.http import request
+from odoo.tools import file_open
+
+
+class RtcController(http.Controller):
+    @http.route("/mail/rtc/session/notify_call_members", methods=["POST"], type="json", auth="public")
```
Comment (oomsveta) [inline addons/mail/controllers/discuss/rtc.py:11]: to be renamed

**Review by oomsveta — APPROVED** [review]
move ok 👌

**Comment by seb-odoo** [comment]
@robodoo override=ci/security

Code move. `body` is sanitized when saved as field.

**Comment by seb-odoo** [comment]
@robodoo r+

**Pull request closed without merging**